src/components/composer/composer/abstract_composer_store.ts

```diff
@@ -487,6 +513,145 @@ export abstract class AbstractComposerStore extends SpreadsheetStore {
+        for (let i = 0; i < this.currentTokens.length; i++) {
+          this.currentTokens[i].isBlurred = !this.isTokenInCursorFormulaContext(
+            this.currentTokens[i],
+            previousSymbolAtCursor
+          );
+        }
+        return;
+      }
+    }
+  }
+
+  private getParenthesesCodeAfterCursor(): string {
+    // we always look at the code associated with the token located after the cursor.
+    // This code is the same as the 'tokenAtCursor' except in the case of a closing parenthesis.
+    // In this case we look at the code located one degree below in the parentheses tree
+    const code = this.tokenAtCursor?.parenthesesCode || "";
+    if (this.tokenAtCursor?.type === "RIGHT_PAREN") {
+      return code.slice(0, -1) || "";
+    }
+    return code;
+  }
+
+  private isTokenInCursorFormulaContext(
+    token: EnrichedToken,
+    // we refer to the previous symbol parenthesesCode and not directly the
+    // parenthesesCode of the token at the cursor because we don't want to
+    // match cases where the token at the cursor is between parentheses which
+    // are not function parentheses
+    previousSymbolAtCursor: EnrichedToken
+  ): boolean {
+    return (token.parenthesesCode || "").startsWith(previousSymbolAtCursor.parenthesesCode || "");
+  }
+
+  private computeTokensRelatedToCursor() {
+    let doesTokenMatchCursorToken: (token: EnrichedToken) => boolean = () => false;
+    const tokenAtCursor = this.tokenAtCursor;
+
+    if (tokenAtCursor) {
+      switch (tokenAtCursor.type) {
+        case "LEFT_PAREN":
+        case "RIGHT_PAREN":
+          if (tokenAtCursor.parenthesesCode !== "") {
+            doesTokenMatchCursorToken = (token) => {
+              return (
+                ["LEFT_PAREN", "RIGHT_PAREN"].includes(token.type) &&
+                token.parenthesesCode === tokenAtCursor.parenthesesCode
+              );
+            };
+          }
+          break;
+        case "REFERENCE":
+          const { xc, sheetName } = splitReference(tokenAtCursor.value);
+          const currentSheetName = this.getters.getSheetName(this.sheetId);
+          doesTokenMatchCursorToken = (token) => {
+            const { xc: tXc, sheetName: tSheetName } = splitReference(token.value);
+            return (
+              xc === tXc && (sheetName || currentSheetName) === (tSheetName || currentSheetName)
+            );
+          };
+          break;
+      }
+    }
+
+    let findAtLeastOneMatch: boolean = false;
+    for (let i = 0; i < this.currentTokens.length; i++) {
+      if (
+        doesTokenMatchCursorToken(this.currentTokens[i]) &&
+        this.currentTokens[i] !== this.tokenAtCursor
+      ) {
+        findAtLeastOneMatch = true;
+        this.currentTokens[i].isRelatedToCursorToken = true;
+        continue;
+      }
+      this.currentTokens[i].isRelatedToCursorToken = false;
+    }
+
+    if (findAtLeastOneMatch) {
+      this.tokenAtCursor!.isRelatedToCursorToken = true;
+    }
```

Comment:
```suggestion
    for (let i = 0; i < this.currentTokens.length; i++) {
      if (
        doesTokenMatchCursorToken(this.currentTokens[i]) &&
        this.currentTokens[i] !== this.tokenAtCursor
      ) {
        this.tokenAtCursor!.isRelatedToCursorToken = true;
        this.currentTokens[i].isRelatedToCursorToken = true;
        continue;
      }
      this.currentTokens[i].isRelatedToCursorToken = false;
    }
```
can be simplified a bit. But it looks way better now :D

Reply:
indeed, thx